File: backend/app/models/teacher.py
from sqlalchemy import Column, Integer, String, ForeignKey
from app.database import Base

class Teacher(Base):
    __tablename__ = "teachers"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="SET NULL"), unique=True)
    name = Column(String(100), nullable=False)
    department = Column(String(100))
    max_lectures_per_day = Column(Integer, default=4)

File: SmartTimetable/src/SmartTimetable.Core/Entities/Teacher.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartTimetable.Core.Entities;

[Table("teachers")]
public class Teacher
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("user_id")]
    public int? UserId { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [MaxLength(100)]
    [Column("department")]
    public string? Department { get; set; }

    [Column("max_lectures_per_day")]
    public int? MaxLecturesPerDay { get; set; }

    // Navigation
    [ForeignKey("UserId")]
    public User? User { get; set; }
}
